### PR TITLE
Test user context

### PR DESCRIPTION
**Pull request by nalanart**
Created an actions/user.js file to dispatch from:

moved authenticate() from the useEffect inside the context provider to actions/user.js so that it can be used in each page to retrieve user info on refresh.

### CODE REVIEW

File: client/src/layout/Mention.js [hunk 1]
```diff
@@ -92,16 +93,16 @@ const Mention = ({ mention }) => {
       <CardContent className={classes.content}>
         <Box component="div" className={classes.titleBox}>
           <Typography variant="h6" gutterBottom>
-            {mention.title.substring(0, indexK)}
+            {/* {mention.title.substring(0, indexK)}
             <span style={{ color: "#536dfe" }}>{keyword}</span>
-            {mention.title.substring(indexK + keyword.length)}
+            {mention.title.substring(indexK + keyword.length)} */}
```
Comment (dproc96) [inline client/src/layout/Mention.js:98]: If you're commenting this out, can it just be removed?

File: client/src/layout/SettingCompany.js [hunk 2]
```diff
@@ -57,6 +59,7 @@ const useStyles = makeStyles((theme) => ({
 
 const SettingCompany = () => {
   const classes = useStyles();
+  const { user } = useContext(UserContext);
```
Comment (dproc96) [inline client/src/layout/SettingCompany.js:62]: If I reload this page, it wipes out the autofilled information

File: client/src/pages/Signup.js [hunk 3]
```diff
@@ -95,30 +89,34 @@ const Signup = () => {
...
   };
 
+  useEffect(() => {
+    // is this redirect needed?
+    authenticate(dispatch);
+    if (isAuthenticated) {
+      history.push("/");
+    }
+  }, [isAuthenticated]);
+
```
Comment (dproc96) [inline client/src/pages/Signup.js]: If you're redirecting in the login method I don't think you need it here as well

File: server/controllers/userController.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ const signUp = async (req, res) => {
       const token = generateToken(user._id);
       res.cookie("dolphinToken", token, {
         maxAge: 3600000,
-        sameSite: "none",
+        // sameSite: "none",
```
Comment (dproc96) [inline server/controllers/userController.js]: If you're commenting out this code, can it just be deleted?

**Review by dproc96 — COMMENTED** [review]
This looks good and it's mostly working as expected. Found one weird edge case and had a couple comments on the code

File: server/controllers/userController.js (hunk 4 above)
Comment (nalanart) [reply server/controllers/userController.js]: Removed

File: client/src/layout/Mention.js (hunk 1 above)
Comment (nalanart) [reply client/src/layout/Mention.js:98]: This will be used later to highlight search term

File: client/src/layout/SettingCompany.js (hunk 2 above)
Comment (nalanart) [reply client/src/layout/SettingCompany.js:62]: Fixed

**Review by dproc96 — APPROVED** [review]
This is working as expected, check with the team on commented out code, it's generally not good practice to put commented out code into dev even if you plan on using it later because it can contribute to software entropy and negatively impact readability

But if Jess and the team sign off on that I think this is good to go

**Pull request merged**